hal/src/hal-filesystem.ads

```diff
@@ -116,12 +153,22 @@ package HAL.Filesystem is
                         Entry_Number : Positive;
                         Dir_Entry    : out Directory_Entry)
                         return Status_Kind is abstract;
+   --  Read the Entry_Number'th entry in This. If it exists, store its
+   --  details in Dir_Entry and return Status_Ok. Otherwise, return
+   --  No_Such_File_Or_Directory.
 
    function Entry_Name (This         : in out Directory_Handle;
                         Entry_Number : Positive)
                         return Pathname is abstract;
+   --  Return the absolute path (full name) for the Entry_Number directory
+   --  entry. Return an empty path if this entry does not exist.
+   --
+   --  ??? In a virtual file system, this full path will actually be relative
+   --  to the root directory of the relevant file system. This makes VFS
+   --  traversal cumbersome to write.
```

Comment:
I don't think we want to return the full path here, but just the name of the node.